### PR TITLE
[TRTLLM-16614][chore] Remove more TRT leftovers

### PR DESCRIPTION
**Pull request by Funatiq**
## Dev Engineer Review

- Removed legacy engine-directory, workspace, and build-stat tracking from LLM and model-loader construction.
- Changed `CachedModelLoader.__call__` to return `Optional[Path]`. It returns `None` for `AutoDeployLlmArgs`.
- Removed the public `LlmBuildStats` class and related serialization whitelist entries.
- Updated executor wiring to pass the cached Hugging Face model directory, or `None` for multimodal models.
- Removed unused multi-LoRA integration helpers. This is an unrelated scope change that may affect external test consumers.
- Review severity counts are unavailable.

## QA Engineer Review

- Modified `tests/unittest/llmapi/test_llm_utils.py`.
- Added coverage for Torch model-directory loading and `AutoDeployLlmArgs` returning `None`.
- Added executor-construction tests for `_TorchLLM` and `MultimodalEncoder` argument wiring.
- The supplied summary does not identify test-list entries in `test-db/` or `qa/`.
- Coverage verdict: sufficient for the changed loader and executor paths; follow-up is recommended for removed multi-LoRA integration helpers.

## Per-File QA Perspective

- `tensorrt_llm/llmapi/llm.py`: Verify executor construction without an engine path and logprob behavior without automatic TensorRT logits enabling.
- `tensorrt_llm/llmapi/llm_utils.py`: Verify the reduced loader APIs, model-directory return value, and `None` result for AutoDeploy.
- `tensorrt_llm/llmapi/mm_encoder.py`: Verify multimodal executor construction with `None` as the model argument.
- `tensorrt_llm/llmapi/serialization.py`: Verify removed loader and build-stat classes are no longer accepted by the deserialization whitelist.
- `tests/integration/defs/common.py`: Removed multi-LoRA test helpers and related imports. Verify no remaining integration test references require these helpers.
- `tests/unittest/llmapi/test_llm_utils.py`: Covers loader return values and executor argument wiring. Test-list registration is not identified in the supplied evidence.

## Description

See commit messages for details.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/llmapi/llm.py [hunk 1]
```diff
@@ -2067,7 +2048,7 @@ def _build_model(self):
         # TODO: revisit gather_context_logits
         return_logits = self.args.gather_generation_logits
         self._executor = self._executor_cls.create(
-            self._engine_dir,
+            None,
```
[the remark this answers is not in the document]
Comment (Funatiq) [reply tensorrt_llm/llmapi/llm.py:2051]: Fixed in commit 89d40a9.

**Review by mikeiovine — APPROVED** [review]
Stamp on behalf of runtime devs, delegating proper review to @NVIDIA/trt-llm-models-devs; please ping me if you think this is not accurate

File: tensorrt_llm/llmapi/llm.py [hunk 2]
```diff
@@ -1488,15 +1483,6 @@ def _prepare_sampling_params(
...
-        # auto enable context and/or generation logits flags, as they are required by logprob computation for TRT backend.
-        if self.args.backend not in ["pytorch", "_autodeploy"]:
-            if sampling_params.prompt_logprobs and not sampling_params.return_context_logits:
-                sampling_params.return_context_logits = True
-                sampling_params._context_logits_auto_enabled = True
-            if sampling_params.logprobs is not None and not sampling_params.return_generation_logits:
-                sampling_params.return_generation_logits = True
-                sampling_params._generation_logits_auto_enabled = True
-
         if sampling_params._stream_interval is None:
```
Comment (BowenFu) [inline tensorrt_llm/llmapi/llm.py:1486]: Nice catch that this auto-enable block was already unreachable — `TorchLlmArgs.backend` is `Literal["pytorch"]` and `CachedModelLoader.__call__` raises for anything else, so nothing could take the TRT branch.

One thing that falls out of deleting it: `SamplingParams._context_logits_auto_enabled` and `_generation_logits_auto_enabled` now have no writer anywhere in the tree, which makes `_need_return_context_logits` / `_need_return_generation_logits` equal to the public flags they wrap. Their own comment says "Used in logprobs calculation in TRT flow ... can be deprecated after migration to PyTorch backend", so they look like exactly the leftovers this PR is after. Same story for `ModelLoader`'s `workspace` argument — `self._workspace` is now only assigned, never read.

Non-blocking either way; happy to see it as a follow-up if you'd rather keep this one tight.

**Review by jieli-matrix — APPROVED** [review]

**Review by lori-ren — APPROVED** [review]
Thanks for the efforts! Is there a timeline or an EPIC for follow-up removal of TRT leftovers?

File: tensorrt_llm/llmapi/llm.py (hunk 2 above)
Comment (Funatiq) [reply tensorrt_llm/llmapi/llm.py:1486]: Good catch. Will do it as a follow-up.
